### PR TITLE
Added test to examples/raw_upload

### PR DESCRIPTION
**Pull request by alanstoate**
Added a test for raw_upload

### CODE REVIEW

**Comment by alanstoate** [comment]
Not too sure why the checks failed

**Comment by adwhit** [comment]
"It's not you, it's me" - master is failing with the same error, probably due to #235

File: examples/raw_upload/src/main.rs [hunk 1]
```diff
@@ -21,3 +21,35 @@ fn index() -> &'static str {
...
+    use rocket::http::Method::Post;
+    use rocket::response::NamedFile;
+    use std::io::Read;
+
+    #[test]
+    fn test_raw_upload() {
+        let text_to_upload = "Text to be uploaded!".to_string();
+
+        // Upload the body
+        let rocket =
```
Comment (SergioBenitez) [inline examples/raw_upload/src/main.rs]: The style here isn't consistent. It should look like:

```
let rocket = rocket::ignite()
    .mount(...);
```

File: examples/raw_upload/src/main.rs [hunk 2]
```diff
@@ -21,3 +21,35 @@ fn index() -> &'static str {
...
+            rocket::ignite().mount("/", routes![super::index, super::upload]);
+        let mut request = MockRequest::new(Post, "/upload")
```
Comment (SergioBenitez) [inline examples/raw_upload/src/main.rs]: Add an empty line above here.

File: examples/raw_upload/src/main.rs [hunk 3]
```diff
@@ -21,3 +21,35 @@ fn index() -> &'static str {
...
+            .header(ContentType::Plain)
+            .body(&text_to_upload);
+        let response = request.dispatch_with(&rocket);
```
Comment (SergioBenitez) [inline examples/raw_upload/src/main.rs]: An empty line above here.

File: examples/raw_upload/src/main.rs [hunk 4]
```diff
@@ -21,3 +21,35 @@ fn index() -> &'static str {
 fn main() {
     rocket::ignite().mount("/", routes![index, upload]).launch();
 }
+
+#[cfg(test)]
+mod test {
+    use super::rocket;
```
Comment (SergioBenitez) [inline examples/raw_upload/src/main.rs]: Is this actually needed?

File: examples/raw_upload/src/main.rs [hunk 5]
```diff
@@ -21,3 +21,35 @@ fn index() -> &'static str {
...
+    use rocket::testing::MockRequest;
+    use rocket::http::{Status, ContentType};
+    use rocket::http::Method::Post;
```
Comment (SergioBenitez) [inline examples/raw_upload/src/main.rs]: Just import `Method` above and use `Method::Post` in the tests.

File: examples/raw_upload/src/main.rs [hunk 6]
```diff
@@ -21,3 +21,35 @@ fn index() -> &'static str {
...
+            .body(&text_to_upload);
+        let response = request.dispatch_with(&rocket);
+        assert_eq!(response.status(), Status::Ok);
+
+        // Ensure we find the body in the /tmp/upload.txt file
+        let mut found_text = String::new();
+        let bytes_read = NamedFile::open("/tmp/upload.txt")
+            .and_then(|mut f| f.read_to_string(&mut found_text));
+
+        assert_eq!(bytes_read.expect("Unable to Read"), text_to_upload.capacity());
```
Comment (SergioBenitez) [inline examples/raw_upload/src/main.rs]: Why are we checking against `capacity` and not `len`? Also, doesn't the next test for equality encompass this test? I think we can kill this line.

File: examples/raw_upload/src/main.rs [hunk 7]
```diff
@@ -21,3 +21,35 @@ fn index() -> &'static str {
...
+            rocket::ignite().mount("/", routes![super::index, super::upload]);
+        let mut request = MockRequest::new(Post, "/upload")
+            .header(ContentType::Plain)
+            .body(&text_to_upload);
+        let response = request.dispatch_with(&rocket);
+        assert_eq!(response.status(), Status::Ok);
+
+        // Ensure we find the body in the /tmp/upload.txt file
+        let mut found_text = String::new();
+        let bytes_read = NamedFile::open("/tmp/upload.txt")
```
Comment (SergioBenitez) [inline examples/raw_upload/src/main.rs]: Is there a reason we're not using `File` from `std::fs` here?

**Comment by SergioBenitez** [comment]
@alanstoate Waiting on a few changes here. :)

**Comment by SergioBenitez** [comment]
There was brief experiment with rewriting all of history that has now concluded. GitHub seems to have barfed here. Can you please open a new PR with your changes if you'd still like them reviewed? Thanks!

**Pull request closed without merging**